p4-16/psa/PSA.mdk

```diff
@@ -86,13 +91,30 @@ controls.
 When instantiating the `main` `package` object, the instances
 corresponding to the programmable blocks are passed as arguments.
 
+# Naming conventions
+
+In this document we use the following naming conventions:
+
+- Types are named using CamelCase followed by `_t`. For example, `PortId_t`.
+- Control types and extern object types are named using CamelCase. For
+  example `IngressParser`.
+- Struct types are named using lower case words separated by `_`
+  followed by `_t`. For example `psa_ingress_input_metadata_t`.
+- Actions, extern methods, extern functions, headers, structs, and
+  instances of controls and externs start with lower case and words
+  are separated using `_`. For example `send_to_port`.
+- Enum members, const definitions, and #define constants are all
+  caps, with words separated by `_`. For example 'PORT_CPU'.
+
+Architecture specific metadata (e.g. structs) are prefixed by `psa_`.
+
```

Comment:
Great stuff! Do we also want to define a prefix?

Reply:
You mean for all the other names? I thought about putting it out for debate, but it seems a bit too much. I would favor a "namespace" rather than prefixing each control type with PSA. However, it is certainly something we can debate in the next meeting.

Reply:
Do we have support for namespaces? Namespaces are definitely cleaner than prefixes, but then we need to have it.

Reply:
So, I suggest we discuss it on Monday. The proposal is to add and consistently use psa_ or PSA_ prefix (depending on the particular naming convention).